source/extensions/tracers/datadog/time_util.cc

```diff
@@ -0,0 +1,27 @@
+#include "source/extensions/tracers/datadog/time_util.h"
+
+#include <chrono>
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+datadog::tracing::TimePoint estimateTime(SystemTime wall) {
+  return estimateTime(wall, datadog::tracing::default_clock);
+}
+
+datadog::tracing::TimePoint estimateTime(SystemTime wall, const datadog::tracing::Clock& clock) {
+  auto point = clock();
+  auto elapsed = point.wall - wall;
```

Comment:
I often feel like working with abstract integer-like types that might be negative that it is more obviously correct to compare the two integer-like vars together, rather than subtracting them and comparing that variable  to 0, which might have unexpected types depending on signed-ness of the type